src/Raven.Server/Web/Operations/OperationsHandler.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Net;
+using System.Threading.Tasks;
+using Raven.Server.Documents;
+using Raven.Server.Routing;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+
+namespace Raven.Server.Web.Operations
+{
+    public class OperationsHandler : DatabaseRequestHandler
+    {
+
+        [RavenAction("/databases/*/operations/next-operation-id", "GET")]
+        public Task GetNextOperationId()
```

Comment:
I was struggling with proper solution for this and having operation id was more reasonable solution. 

In v4.0 I don't pull operation id each 500 ms, instead I use changes API (watch operation by id). In following code snippet:
1) Send update by index command -> returns operation id
2) Send watch-operation with arg
3) Wait for notification

We get race condition, because operation might be completed before sending watch-operation command. 

This is why I create operation object and fetch operation id. Then I subscribe to notifications and execute operation passing operation id as parameter. Then we always get notification about completion/progress. This notification is automatically disposed when operation is finished.